src/app/core/models/business.model.ts

```diff
@@ -1,35 +1,148 @@
-  RESTAURANT = 'Restaurant',
+// export enum CATEGORY_BUSINESS {
+//   RESTAURANT = 'Restaurant',
+// }
+
+export enum BUSINESS_PAYMENT_TYPES {
+  NOTHING = 'NOTHING',
+  CREDIT = 'CREDIT',
+  DEBIT = 'DEBIT',
+  CASH = 'CASH',
+  CRYPTO = 'CRYPTO'
 }
 
-export function values() {
-  return Object.keys(CATEGORY_BUSINESS).filter(
-    (type) => isNaN(type as any) && type !== 'values'
-  );
+export enum BUSINESS_SOCIAL_LINKS {
+  FACEBOOK = 'FACEBOOK',
+  TWITTER = 'TWITTER',
+  INSTAGRAM = 'INSTAGRAM'
 }
 
-export class Business {
+export enum BUSINESS_STATUSES {
+  PENDING = 'PENDING',
+  ACCEPTED = 'ACCEPTED',
+  REFUSED = 'REFUSED'
+}
+
+export enum BUSINESS_DAYS_WEEK {
+  MONDAY = 'MONDAY',
+  TUESDAY = 'TUESDAY',
+  WEDNESDAY = 'WEDNESDAY',
+  THURSDAY = 'THURSDAY',
+  FRIDAY = 'FRIDAY',
+  SATURDAY = 'SATURDAY',
+  SUNDAY = 'SUNDAY'
+}
+
+// export function businessPaymentTypesValues() {
+//   return Object.keys(BUSINESS_PAYMENT_TYPES).filter(
+//     (type) => isNaN(type as any) && type !== 'values'
+//   );
+// }
+
+export class Category {
   id?: number;
-  name: number;
-  description: string;
-  website: string;
-  email: string;
-  acceptedAt: string;
-  slogan?: string;
-  language: LANGUAGE;
-  maximalCapacity: number;
-  logo?: string;
-  createdAt: string;
-  updatetedAt?: string;
-  deletedAt?: string;
-  category: CATEGORY_BUSINESS;
-}
-
-export class Restaurant extends Business {
+  name: string;
+}
+
+export class BusinessHour {
   id?: number;
+  businessId: number;
+  day: BUSINESS_DAYS_WEEK;
+  closingTime: string;
+  openingTime: string;
+}
+
+export class Business {
+  protected id?: number;
+  protected phoneId?: number[];
+  protected categoryId?: number;
+  protected restaurantId?: number;
+  @required({message: 'admin.business.message_errors.name_required'})
+  @trim()
+  protected name: string;
+  // @alphaNumeric({message: 'admin.business.message_errors.description_alpha'})
+  @prop()
+  @trim()
+  protected description: string;
+  // @alphaNumeric({message: 'admin.business.message_errors.note_alpha'})
+  @prop()
+  @trim()
+  protected note: string;
+  @url({message: 'admin.business.message_errors.url_valid'})
+  @trim()
+  protected website: string;
+  @trim()
+  @email({message: 'admin.business.message_errors.email_valid'})
+  @required({message: 'admin.business.message_errors.email_required'})
+  protected email: string;
+  @alphaNumeric({message: 'admin.business.message_errors.name_required'})
+  @trim()
+  protected slogan?: string;
+  @required({message: 'admin.business.message_errors.language_required'})
+  protected language: BUSINESS_LANGUAGE;
+  @numeric({message: 'admin.business.message_errors.numeric_valid'})
+  protected capacity?: number;
+  protected createdAt: string;
+  protected updatetedAt?: string;
+  protected acceptedAt: string;
+  protected deletedAt?: string;
+  @required({message: 'admin.business.message_errors.payment_type_required'})
+  protected paymentType: BUSINESS_PAYMENT_TYPES;
```

Comment:
```suggestion
  protected paymentType: BUSINESS_PAYMENT_TYPES.PENDING;
```
N'est ce pas plus censé?

Reply:
nop parce que la c'est un type bien plus precis c'est un string de toute facon mais une valeur bien plus précise